src/interface/report/EventParser.js

```diff
@@ -91,12 +91,11 @@ class EventParser extends React.PureComponent {
     return parser;
   }
   makeEvents(parser) {
-    const { bossPhaseEvents, events } = this.props;
-    let combinedEvents = bossPhaseEvents ? [...bossPhaseEvents, ...events] : events;
+    let { events } = this.props;
     // The events we fetched will be all events related to the selected player. This includes the `combatantinfo` for the selected player. However we have already parsed this event when we loaded the combatants in the `initializeAnalyzers` of the CombatLogParser. Loading the selected player again could lead to bugs since it would reinitialize and overwrite the existing entity (the selected player) in the Combatants module.
-    combinedEvents = combinedEvents.filter(event => event.type !== 'combatantinfo');
-    combinedEvents = parser.normalize(combinedEvents);
-    return combinedEvents;
+    events = events.filter(event => event.type !== 'combatantinfo');
+    events = parser.normalize(events).sort((a,b) => a.timestamp - b.timestamp); //sort now normalized events
```

Comment:
please leave a comment to explain why (I can see you're sorting it now)

Reply:
added a comment to the code, basically the reason is that after normalizing, the new events (like the fabricated prepull events etc) would get "mixed in" with the cast events "kept" from before the filter (to allow for cooldown information to remain accurate), additionally since buff events from before the filter get moved to the beginning of the filter, whereas cast events do not, this led to events being "out of order" even just before the phase. Granted the only events impacted by this were events before the filter / counted as prepull, so it didn't make a big impact, but I wanted to fix it anyways even if just for the events tab. Figured the easiest way to fix that is to just sort the full events list one last time before actually handing it off to the parser